### PR TITLE
feat: update Task status based on the default project's status in the Projects child table

### PR DESCRIPTION
**Pull request by vishdha**
TASK ID: https://bloomstack.com/desk#Form/Task/TASK-2021-00288

Gif: 
Peek 2021-09-15 16-28

### CODE REVIEW

File: erpnext/projects/doctype/task/task.py [hunk 1]
```diff
@@ -120,9 +121,11 @@ def validate_default_project(self):
 				frappe.throw(_("There can be only one default project, found {0}.").format(len(default_project)))
 			else:
 				self.project = default_project[0].project
+				self.status = default_project[0].status
 		elif not len(self.projects):
```
Comment (neilLasrado) [inline erpnext/projects/doctype/task/task.py]: you can remove this elif as we are making projects mandatory

File: erpnext/projects/doctype/task/task.json [hunk 2]
```diff
@@ -116,11 +116,12 @@
    "fieldtype": "Select",
    "in_list_view": 1,
    "in_standard_filter": 1,
-   "label": "Status",
+   "label": "Default Status",
```
Comment (neilLasrado) [inline erpnext/projects/doctype/task/task.json]: leave the label as Status, don't change it, instead add a description that the status of the default project will be set.

**Review by neilLasrado — CHANGES_REQUESTED** [review]

**Review by neilLasrado — DISMISSED** [review]

File: erpnext/projects/doctype/task/task.json [hunk 3]
```diff
@@ -112,6 +112,7 @@
   },
   {
    "bold": 1,
+   "description": "the status of the default project will fetch from the projects table.",
```
Comment (Anurag810) [inline erpnext/projects/doctype/task/task.json]: It should be like .... "**T**he......default **P**roject........**P**rojects table". 
As the label of the table is Projects and Project is also a DocType.

**Review by Anurag810 — DISMISSED** [review]

File: erpnext/projects/doctype/task/task.json [hunk 4]
```diff
@@ -112,6 +112,7 @@
   },
   {
    "bold": 1,
+   "description": "The status of the default Project will fetch from the Projects table.",
```
Comment (sahil28297) [inline erpnext/projects/doctype/task/task.json]:
```suggestion
   "description": "The status of the default Project is fetched from the Projects table.",
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**